File: Java/Medium/WrongAnswer/minimum-garden-perimeter-to-collect-enough-apples_2.java
class Solution {
    public long minimumPerimeter(long neededApples) {
        
        long apples = 0;
        long i = 0;
        while (apples < neededApples) {
            i++;
            apples += (i * 2 + 1) * (i * 2 + 1) * 2 + (i * 2) * (i * 2) * 2 + (i * 2 + 1) * (i * 2) * 4;
        }
        return i * 8;
    }
}

File: C++/Medium/WrongAnswer/minimum-garden-perimeter-to-collect-enough-apples_2.cpp
class Solution {
public:
    long long minimumPerimeter(long long neededApples) {
        
        long long n = 1;
        while (n * (n + 1) * (2 * n + 1) < 12 * neededApples) {
            ++n;
        }
        return 8 * n;
    }
};